src/components/Tokens/TokenTable/TokenTable.tsx

```diff
@@ -95,7 +95,7 @@ export default function TokenTable() {
 
   /* loading and error state */
   if (loading && (!tokens || tokens?.length === 0)) {
-    return <LoadingTokenTable rowCount={PAGE_SIZE} />
+    return <LoadingTokenTable rowCount={Math.min(PAGE_SIZE, maxFetchable)} />
```

Comment:
It is unclear that TokenDataContainer is capped at PAGE_SIZE, so this limitation looks ad-hoc. Can you point out where that is done? Will tokens.length be maxFetchable, or will it be Math.min(PAGE_SIZE, maxFetchable)? Can maxFetchable be capped to PAGE_SIZE, so that this logic can be silo'ed to useTopTokens and not need to be duplicated on this line?

Reply:
I think I get what you mean, the ad-hoc nature makes it unclear why we are doing this. 

I split out this logic into "loadingRowsCount" useMemo fn in TopToken.ts and added a comment. hope it makes it more clear now?